frontend/src/lib/components/questions/LikertResponseButtons.svelte

```diff
@@ -0,0 +1,270 @@
+   *   updated. @default undefined
+   */
+  function dispatchEvent(value?: AnswerOption['key'] | null) {
+    // Use the selected value if no value is supplied
+    value ??= selected;
+    // Only dispatch the event if the value is defined
+    if (value == null) return;
+    dispatch(
+      // Check whether the selected value has changed or not
+      selectedKey != null && value == selectedKey ? 'reselect' : 'change',
+      {id: name, value} as LikertResponseButtonsEventDetail
+    );
+  }
+</script>
+
+<!--
+@component
+Display the radio buttons used for answering Likert and other ordinal, 
+multiple choice questions.
+
+The buttons are rendered as `<input type="radio">` elements contained
+inside a `<fieldset>`. Consider passing an `aria-labelledby` pointing to
+the question or an `aria-label`.
+
+The radio buttons' behaviour is as follows when using a pointer or touch 
+device:
+
+1. Selecting an option dispatches a `change` event 
+2. Clicking on the selected radio button dispatches a `reselect` event
+
+Keyboard navigation works in the following way:
+
+1. `Tab` focuses the whole radio group
+2. The arrow keys change the focused radio button and *select* it at the
+   same time
+3. When the keyboard focus leaves the radio group, either of the events
+   is dispatched, depending on whether value has been changed or not
+4. The event is also dispatched when the user presses the `Space` or 
+   `Enter` key
+
+### Properties
+
+- `name`: The `name` of the radio group. Usually the question's id
+- `options`: The `key`-`label` pairs of the radio buttons
+- `mode`: The same component can be used both for answering the questions 
+  and displaying answers. @default `'answer'`
+- `selectedKey`: The initially selected key of the radio group.
+- `entityKey`: The answer key of the entity in display mode.
+- `entityLabel`: The label for the entity's answer. Be sure to supply this 
+  if `entityKey` is supplied.
+- Any valid attributes of a `<fieldset>` element
+
+### Events
+
+- `reselect`: The user has clicked on the same radio button that
+  was initially selected.
+- `change`: The user has clicked on a different radio button than
+  which was initially selected or there was no selected value initially
+- Both contain the a `detail` object with the properties:
+  - `id`: The `name` of the radio group. Usually the question's id
+  - `value`: The selected key of the radio group.
+
+### Usage
+
+```tsx
+<LikertResponseButtons
+  aria-labelledby="questionHeadingId"
+  name={question.id}
+  options={question.options}
+  selectedKey={$voterAnswers[question.id]}
+  on:change={answerQuestion}
+  on:reselect={doFoo} />
+
+<LikertResponseButtons
+  aria-labelledby="questionHeadingId"
+  name={question.id}
+  options={question.options}
+  mode="display"
+  selectedKey={$voterAnswers[question.id]}
+  entityKey={candidateAnswer}
+  entityLabel={$_('candidate.candidateAnswerLabel')}
+   />
+```
+-->
+
+<fieldset
```

Comment:
If users want to use the keyboard, how should they navigate the answers? Will they use arrow keys or the Tab key?

Reply:
They will use the arrow keys, which is the default way a radio group operates. `Tab` will move focus out of the widget, which is also the default.